### PR TITLE
Actually enforce and use SwiftLint for the toolkit

### PR DESCRIPTION
**Pull request by Matt-Gleich**
## Description

Actually, enforce and use swiftlint for the Toolkit.

## Type of change

- [ ] New script command
- [ ] Bug fix
- [ ] Improvement of an existing script
- [ ] Documentation update
- [x] Toolkit change
- [ ] Other (Specify)

## Screenshot

N/A

## Dependencies / Requirements

N/A

## Checklist

- [x] I have read [Contribution Guidelines](https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Matt-Gleich** [comment]
cc @unnamedd

**Comment by Matt-Gleich** [comment]
In the future, lets add CI for this and Xcode support

**Comment by thomaspaulmann** [comment]
Leaving it to @unnamedd to review since he's the master of the toolkit. Generally looks good to me.

**Comment by unnamedd** [comment]
Sorry guys, I didn't had much time to review it, but I promise I will do it today! 🙏😅

File: Tools/Toolkit/.swiftlint.yml [hunk 1]
```diff
@@ -40,11 +40,10 @@ indentation_width:
 trailing_comma:
```
Comment (unnamedd) [inline Tools/Toolkit/.swiftlint.yml:40]: This file must not be moved out of the repo’s root. To use it, we need to:
1) set the path when we call SwiftLint
or
2) copy from the root to the current folder and then delete after

we don’t want to maintain two files, one in the Toolkit’s folder and one in the root

File: Tools/Toolkit/Makefile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+lint:
```
Comment (unnamedd) [inline Tools/Toolkit/Makefile:1]: These instructions here, must live in the main Makefile (in the root). We need to concentrate all the building/generation/validation process from there.

File: Tools/Toolkit/Sources/ToolkitLibrary/Core/Documentation/Documentation.swift [hunk 3]
```diff
@@ -109,7 +108,7 @@ private extension Documentation {
   func renderMarkdown(for group: Group, headline: Bool = false, leadingPath: String = .empty) -> String {
     var contentString = String.empty
 
-    if group.scriptCommands.count > 0 {
+    if !group.scriptCommands.isEmpty {
```
Comment (unnamedd) [inline Tools/Toolkit/Sources/ToolkitLibrary/Core/Documentation/Documentation.swift:111]: I prefer to keep `.count > 0` or `.isEmpty == false` to keep the good readability.

File: Tools/Toolkit/Sources/ToolkitLibrary/Core/Toolkit/Toolkit+ReadContent.swift [hunk 4]
```diff
@@ -157,8 +157,7 @@ extension Toolkit {
     let valueRange = result.range(withName: "value")
 
     if let key = self.content(of: keyRange, on: content),
-       let value = self.content(of: valueRange, on: content) {
-
+      let value = self.content(of: valueRange, on: content) {
```
Comment (unnamedd) [inline Tools/Toolkit/Sources/ToolkitLibrary/Core/Toolkit/Toolkit+ReadContent.swift:160]: Please, keep the space.

File: Tools/Toolkit/Sources/ToolkitLibrary/Core/Toolkit/Toolkit+ReadContent.swift [hunk 5]
```diff
@@ -101,7 +101,7 @@ extension Toolkit {
     var dictionary: [String: Any] = [:]
     let authors = extractAuthors(from: content, using: results)
 
-    if authors.count > 0 {
+    if !authors.isEmpty {
       dictionary["authors"] = extractAuthors(from: content, using: results)
```
Comment (unnamedd) [inline Tools/Toolkit/Sources/ToolkitLibrary/Core/Toolkit/Toolkit+ReadContent.swift:105]: The same comment I from above: to keep a good readability, please use `.count > 0` or `.isEmpty == false` for the three changes made near to this comment.

**Review by unnamedd — CHANGES_REQUESTED** [review]
@Matt-Gleich thanks a lot for your work here. 

There are some changes that need rollback or improvements. When you finished to address them, please, let me know and I will re-review them!  :)

**Review by thomaspaulmann — CHANGES_REQUESTED** [review]
Looks like @unnamedd requested some changes. I put this back in your queue @Matt-Gleich. Just request another review when everything is ready.

**Comment by unnamedd** [comment]
@thomaspaulmann, Matthew told me this week he didn’t has time to work on this PR due to school. Then, let’s wait him finish all his tasks. I am in touch with him! :)

**Comment by thomaspaulmann** [comment]
Cool, I wasn't aware of it and that's totally fine by me. Just wanted to make sure that the PR is in the proper state. @Matt-Gleich whenever you feel this is ready, give it another spin. Gonna look forward to get the toolkit properly linted :-)

File: .swiftlint.yml [hunk 6]
```diff
@@ -16,6 +16,7 @@ disabled_rules:
   - type_name
   - function_body_length
   - type_body_length
+  - indentation_width
```
Comment (Matt-Gleich) [inline .swiftlint.yml:19]: This is the only way to have that one line to be indented 3 spaces instead of 2.

File: Makefile [hunk 7]
```diff
@@ -20,8 +22,14 @@ build-debug:
...
 	./$(EXECUTABLE_NAME) generate-documentation
 
 gen-docs-and-commit: gen-docs
 	./$(TOOLKIT_PATH)/integration.sh commit
+
+lint:
+	cd $(TOOLKIT_PATH) && swiftlint --config $(SWIFTLINT_CONFIG_PATH) --path $(TOOLKIT_PATH)
+
+fix:
+	cd $(TOOLKIT_PATH) && swiftlint --config $(SWIFTLINT_CONFIG_PATH) autocorrect
```
Comment (Matt-Gleich) [inline Makefile:35]: For some strange reason `--path $(TOOLKIT_PATH)` was not working for me so this is a workaround.

File: .swiftlint.yml (hunk 6 above)
Comment (unnamedd) [reply .swiftlint.yml:19]: 3 spaces? Raycast standard is 2 spaces. Where did you see 3 spaces?

File: Makefile (hunk 7 above)
Comment (unnamedd) [reply Makefile:35]: I will clone your branch and make some tests here, I don't think that change the directory is a good solution. I will let you know later.

**Comment by unnamedd** [comment]
As discussed privately with @Matt-Gleich, we agreed to close this issue. All the changes he'd be done here will be embedded in my improvements for the Toolkit.

**Comment by thomaspaulmann** [comment]
Ok cool. Thanks for keeping us in the loop and great to know that we have a well formatted codebase for the toolkit.

**Pull request closed without merging**